update/Pos/Update/Poll/PollState.hs

```diff
@@ -72,10 +73,16 @@ modifyPollState PollModifier {..} PollState {..} =
     PollState (modifyHashMap pmBVs _psBlockVersions)
               (fromMaybe _psAdoptedBV pmAdoptedBVFull)
               (fromMaybe _psEpochProposers pmEpochProposers)
-              (modifyHashMap pmConfirmed _psConfirmedBVs)
+              (modifyHashMap pmConfirmed _psConfirmedANs)
               (modifyHashMap pmConfirmedProps _psConfirmedProposals)
               (modifyHashMap pmActiveProps _psActiveProposals)
-              _psActivePropsIdx
+              newActivePropsIdx
               (fromMaybe _psSlottingData pmSlottingData)
               _psFullRichmenData
               _psIssuersStakes
+  where
+    (dels, ins) =
+        (HS.fromList . deletions) &&& (HS.fromList . map fst . insertions) $ pmActiveProps
+    newActivePropsIdx =
+        (fmap removeAndAddProps . (`HM.difference` pmDelActivePropsIdx)) _psActivePropsIdx
+    removeAndAddProps hashset = HS.union ins . (flip HS.difference dels) $ hashset
```

Comment:
> Regarding the insertions, I can't figure out from which ApplicationName a given UpId came from, can I?

You can, because `UpdateProposal` cotains `SoftwareVersion` which in turn contains `ApplicationName`.